### PR TITLE
[Rails 5.2] Remove serialize + JSONB calls

### PR DESCRIPTION
**Pull request by NickLaMuro**
When using serialize, if the database column type is JSONB, then using
serialize is not allowed in Rails 5.2.  With it in place, you end up getting an
error like this:

```
Failure/Error: expect { FactoryBot.create(:rate, :price_plan => plan) }.to change(plan.rates, :count).from(0).to(1)

ActiveRecord::AttributeMethods::Serialization::ColumnNotSerializableError:
  Column `screener` of type ActiveRecord::ConnectionAdapters::PostgreSQL::OID::Jsonb does not support `serialize` feature.
  Usually it means that you are trying to use `serialize`
  on a column that already implements serialization natively.
```

This "should" fix that issue.

Links
-----

- Part of the Rails 5.2 effort: https://github.com/ManageIQ/manageiq/issues/20032

### CODE REVIEW

**Review by jrafanie — APPROVED** [review]

**Comment by jrafanie** [comment]
@miq-bot add_label jansa/yes?

**Review by kbrock — APPROVED** [review]
YAY LGTM

**Comment by kbrock** [comment]
@NickLaMuro Sorry...

can you add 1 unit test to set a field, save it, then load it?

That way we can see that this does not work in 5.1 but does work in 5.2

**Comment by NickLaMuro** [comment]
@kbrock sure thing, I will look into doing that tomorrow.

**Comment by NickLaMuro** [comment]
@kbrock actually, I think there are already some specs in place for this, but it is a bit weird since it seems to only raise the error once per model (on autoload) and then works just fine after that.

You can see this in action by running the `spec/models/data_view_spec.rb`, and notice that there is always 3 errors, but they happen at completely different runs based on the `Rspec` seed.  I did start writing some `it 'serializes JSONB fields'` specs for each of the models, but since the failures are only going to show up on the spec that happens to be the first one to load the model, and thus cause the error, seems like these extra specs won't do much good.

Thoughts?

**Comment by jrafanie** [comment]
Yeah, I agree with @NickLaMuro.  I don't know how you write a test for this that reliably works/doesn't work before/after this change.  It might be helpful for initial verification but seems to not have much value in keeping.

**Comment by NickLaMuro** [comment]
Cool.  Works for me.

FYI, I tossed the specs I was going to add over in another branch for reference:

https://github.com/NickLaMuro/manageiq-consumption/tree/rails-5-2-fixes-with-specs

**Comment by kbrock** [comment]
@NickLaMuro so you just said in rails 5.1 the data serializes just fine without the `serialize :column, JSON` ?

if so, then lets merge this now
if not, then lets remember to merge this when we go to 5.2

(determining ^ is why I suggested the test. sorry if I was not clear)

**Comment by NickLaMuro** [comment]
> @NickLaMuro so you just said in rails 5.1 the data serializes just fine without the serialize :column, JSON ?

I did?  Pretty sure I said this in chat:

> @kbrock Does ManageIQ/manageiq-consumption#172 need to be merged only after we go to 5.2?
> @NickLaMuro I _**think**_ so...

That said...

> (determining ^ is why I suggested the test. sorry if I was not clear)

This branch isn't testing with Rails 5.2, but Rails 5.1 (and passes), but I have been testing it locally with Rails 5.2 (using a symlinked core repo) and the tests pass there as well.  I think the current specs prove that it works fine in Rails 5.1, and a cross-repo run ( @jrafanie ? ) should be able to confirm this is working fine on 5.2.

**Comment by NickLaMuro** [comment]
We should probably wait for said cross-repo to run before merging though.

**Comment by kbrock** [comment]
since this works in rails 5-1, can we merge now?

**Comment by NickLaMuro** [comment]
@kbrock 🐑 🇮🇹

**Review by jrafanie — APPROVED** [review]

**Comment by simaishi** [comment]
Jansa backport details:

```
$ git log -1
commit 17200c151daeac285b840c5ef76b3e4b98afce7b
Author: Gregg Tanzillo <gtanzill@redhat.com>
Date:   Wed Apr 29 13:12:06 2020 -0400

    Merge pull request #172 from NickLaMuro/rails-5-2-fixes

    [Rails 5.2] Remove serialize + JSONB calls

    (cherry picked from commit 4346d9b49c2b9b851ca96c6d48c31147a422fd74)
```

**Pull request merged**
